From 08ad2ef8ca8053a1978cd7e83d2d8815320fe7af Mon Sep 17 00:00:00 2001
From: Patrick Yavitz <pyavitz@xxxxx.com>
Date: Mon, 13 Nov 2023 16:08:17 -0500
Subject: [PATCH] sun50i-h5-nanopi-neo-plus2: drop support for memory dynamic
 frequency scaling

The introduction of DRAM devfreq support for Allwinner A64/H5
has made the unit unstable. So drop support for the driver.

https://lore.kernel.org/linux-arm-kernel/a48923b7-12f9-808e-1171-49b826bd5f1c@samsung.com/T/#ma00a3e07248dc7fb2d300b7c9c409f69ffa64c14

Signed-off-by: Patrick Yavitz <pyavitz@xxxxx.com>
---
 .../dts/allwinner/sun50i-h5-nanopi-neo-plus2.dts     | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/arch/arm64/boot/dts/allwinner/sun50i-h5-nanopi-neo-plus2.dts b/arch/arm64/boot/dts/allwinner/sun50i-h5-nanopi-neo-plus2.dts
index 5e07a5352e76..65a447f0086a 100644
--- a/arch/arm64/boot/dts/allwinner/sun50i-h5-nanopi-neo-plus2.dts
+++ b/arch/arm64/boot/dts/allwinner/sun50i-h5-nanopi-neo-plus2.dts
@@ -89,6 +89,18 @@ rfkill_bt {
 	};
 };
 
+&mbus {
+	compatible = "allwinner,sun8i-h3-mbus";
+	reg = <0x01c62000 0x1000>;
+	/delete-property/ reg-names;
+	clocks = <&ccu CLK_MBUS>;
+	/delete-property/ clock-names;
+	#address-cells = <1>;
+	#size-cells = <1>;
+	dma-ranges = <0x00000000 0x40000000 0xc0000000>;
+	#interconnect-cells = <1>;
+};
+
 &cpu0 {
 	cpu-supply = <&vdd_cpux>;
 };
-- 
2.39.2

